src/Castle.Core/DynamicProxy/AssemblyBuilderAccessDefaults.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright 2004-2021 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+namespace Castle.DynamicProxy
+{
+	using System.Reflection.Emit;
+
+	public static class AssemblyBuilderAccessDefaults
+	{
+#if NET
```

Comment:
Note to self: need to research what this conditional compilation symbol is for, and whether it's needed.

Does it guard the `RunAndCollect` enum value, or the `AssemblyLoadContext.GetLoadContext`? If the former, then perhapa the whole type being added here shouldn't be exposed publicly for those TFMs; so I'm suspecting it's probably guarding the latter.

Reply:
It is to guard against usage of AssemblyLoadContext (it is not in netatandard / netfx, only in .net core).

However, it would be nice if the check could be done at runtime (in case of loading netatandard assemblies in a .net core executable). We have usage of reflection based wrappers at Unity for this kind of thing. I can probably do the same here.